make-file-list.sh

```diff
@@ -372,11 +389,14 @@ etc/post-install/01-devices.post
 etc/post-install/03-mtab.post
 etc/post-install/06-windows-files.post
 usr/bin/start
-mingw$BITNESS/bin/pdftotext.exe
-mingw$BITNESS/bin/libstdc++-6.dll
+$MSYSTEM_LOWER/bin/pdftotext.exe
 usr/bin/column.exe
 EOF
 
+[[ -z "$MINIMAL_GIT" && $MSYSTEM_LOWER == mingw* ]] && cat <<EOF
```

Comment:
The `[[ ... ]] &&` syntax is not used anywhere else in this file, therefore it threw me quite a bit. Let's not?

Also, maybe it's better to introduce a new variable like `PDFTOTEXT_FILES`, and set it in an architecture-specific way before using it here.

Reply:
I used it so that I could do `mingw*` (wildcard). `test` can't do that. Do you have a better alternative to this: `if test -z "$MINIMAL_GIT" && (test "$MSYSTEM_LOWER" = "mingw64" || test "$MSYSTEM_LOWER" = "mingw32")`

I'm not sure whether I get the comment regarding `pdftotext`. That executable is supported on all platforms, so no need to introduce architecture-specific logic for it, right?

Reply:
Right, the wildcard. The classical Unix method is to use `case`, like so:

```sh
case "$MINIMAL_GIT,$MSYSTEM_LOWER" in
,mingw*)
  ...
  ;;
esac
```

But that's a _lot_ more verbose, so I guess I'm fine with the `[[` construct.

> I'm not sure whether I get the comment regarding pdftotext.

What I mean is this: https://github.com/git-for-windows/build-extra/pull/437/files#diff-9b50f413340b32f6bc699741e8b09ab52b3e9c7698d787cc26c4c95e78580916L375-R401

```diff
-mingw$BITNESS/bin/pdftotext.exe
-mingw$BITNESS/bin/libstdc++-6.dll
+$MSYSTEM_LOWER/bin/pdftotext.exe
 usr/bin/column.exe
 EOF
 
+if test -z "$MINIMAL_GIT" && (test "$MSYSTEM_LOWER" = "mingw64" || test "$MSYSTEM_LOWER" = "mingw32"); then
+cat <<EOF
+$MSYSTEM_LOWER/bin/libstdc++-6.dll
+EOF
+fi
+
```

Separating the `pdftotext.exe` line from the `libstdc++-6.dll` line in the proposed way won't make it as obvious as the current version that `pdftotext.exe` is the only reason why that `.dll` is required in the first place.

What I would prefer is to have conditionals that add the appropriate lines for pdftotext depending on the platform, _unless_ we're enumerating the files for a MinGit.

So I would define these lines somewhat like this:

```
case $MSYSTEM_LOWER in
mingw*)
  PDFTOTEXT_FILES="$MSYSTEM_LOWER/bin/pdftotext.exe
$MSYSTEM_LOWER/bin/libstdc++-6.dll"
  ;;
*)
  # In the clang version, we do not need the libstdc++ DLL
  PDFTOTEXT_FILES="$MSYSTEM_LOWER/bin/pdftotext.exe
  ;;
esac
```

and then replace the current mentions of these files with `$PDFTOTEXT_FILES`.

Reply:
Ah, I thought that `libstdc++-6.dll` was used for other things as well. Have just updated the commit and addressed the other feedback from this PR as well. Thanks!